### DISCUSSION TITLE
Stacking different inputs for the same graph to a batch dimension

### DISCUSSION THREAD

**Discussion by knshnb** [pyg-team/pytorch_geometric#8156] [Q&A] [answered]
Hi! I would like to handle multiple different inputs (e.g., node features) of the same graph structure. I can handle them as different graphs, but it might be a little inefficient. Would it be possible to pass batched inputs for a single graph, such as `(n_node, n_batch, n_feat)`, to graph conv?

**Comment by rusty1s** [comment, marked as answer]
There exists a few layers in PyG that support this kind of structure (see the "Static Graph" column [here](https://pytorch-geometric.readthedocs.io/en/latest/cheatsheet/gnn_cheatsheet.html). With it, you can do
```python
x = ... # [batch_size, num_nodes, in_channels]
edge_index = ... # edges of a single graph

out = conv(x, edge_index)  # [batch_size, num_nodes, out_channels]

**Reply by knshnb** [reply]
Thanks for the quick response! That answered my question.
